### ISSUE TITLE
Outline actions, as needed, regarding config for MC2 (moving to data models repo)

### ISSUE THREAD

**Issue by aclayton555** [mc2-center/data-models#42]
See related ticket in HTAN: https://github.com/ncihtan/data-models/issues/320

**Comment by Bankso** [comment]
As of 1.4.24, the [Sage data_curator_config](https://github.com/Sage-Bionetworks/data_curator_config) repository has been refactored and some of the file structure/content has been modified. 
- prod is now the head repo, so any changes made to the config must be merged to prod
- dca_config is now formatted as JSON (was previously CSV) and is split on a per-dcc basis
- [tenants.json](https://github.com/Sage-Bionetworks/data_curator_config/blob/prod/tenants.json) was added

Given the refactor, there are options for how DCC-specific information associated with the DCA is managed. Specifically, "dca-template-config.json" can be stored separately from the data_curator_config repo and the path can be provided through "dca_config.json". This simplifies the process of updating which templates a DCC wants to make available through the DCA UI. 

Suggested actions to shift "dca-template-config.json" to data-models:
1. Sync the [MC2 fork of data_curator_config](https://github.com/mc2-center/data_curator_config/tree/main) with the Sage Bionetworks fork
2. From the prod branch, copy the existing [MC2-specific "dca-template-config.json"](https://github.com/Sage-Bionetworks/data_curator_config/blob/prod/MC2/dca-template-config.json) to a new branch in the MC2 data-models repo
3. Update the data-models "dca-template-config.json" to match current list of templates that should be shown in the DCA (skip any automation for now)
4. Make a PR to merge the branch to dev/main branch of data-models

With the "dca-template-config.json" now in the data-models repo, data_curator_config **prod** branch needs to be updated (use new branches and make PRs to add changes):
1. In the MC2 fork of data_curator_config, update the prod branch "dca_config.json" with current data model release and template config URLs [(see the file content descriptions here)](https://github.com/Sage-Bionetworks/data_curator_config/tree/prod?tab=readme-ov-file#project-configurations-for-the-data-curator-app). Note that the URLs must correspond to raw files.
2. Ensure the manifest templates in the ["MC2/templates"](https://github.com/mc2-center/data_curator_config/tree/main/MC2/templates) folder (note that this example is from the main branch, be sure to work in the prod branch!) match those listed in "dca-template-config.json". If the templates do not match or need to be updated, check data-models to see if the templates associated with the latest release have been added. If not, or if you would prefer to generate them yourself, you can use `schematic manifest get`
3. Remove "MC2/dca-template-config.json" from the repo
4. Merge updates into the prod branch of the MC2 data_curator_config fork
5. Create a PR to merge the MC2 fork of data_curator_config/prod with Sage Bionetworks prod
6. Once the PR is merged, the DCA should point to config files stored in the data-models repo. Validate by checking templates available in DCA dropdown

Moving forward, the following actions will need to be performed as part of our **data model release workflow**:
All updates should be made in a new branch
1. Ensure the manifest templates in the ["MC2/templates"](https://github.com/mc2-center/data_curator_config/tree/main/MC2/templates) folder (note that this example is from the main branch, be sure to work in the prod branch!) match expectations. If a template is missing or needs to be updated, check data-models to see if the templates associated with the latest model release have been added. If not, or if you would prefer to generate them yourself, you can use `schematic manifest get`. 
2. Review and update "dca-template-config.json" in the MC2 data-models repo (can be automated, do manually for now) 
3. Update data model URL in "dca_config.json"
4. Merge updates into the prod branch of MC2/data_curator_config
5. Make a PR to merge the prod branch of MC2/data_curator_config with the Sage Bionetworks prod branch
6. Validate any changes were propagated to DCA

**Comment by Bankso** [comment]
For the MC2 data model (version 23.12.1) the following contributor-focused templates should be listed in "dca-template-config.json" 
- Dataset View
- Educational Resource
- Publication View
- Tool View
- Biospecimen
- NanoString GeoMx DSP Level 1
- NanoString GeoMx DSP Level 2
- NanoString GeoMx DSP Level 3
- NanoString GeoMX ROI Segment Annotation
- NanoString GeoMx DSP Imaging Level 2

The listed templates should only be changed if we add/remove a model or change names.

**Comment by Bankso** [comment]
@aditigopalan and @aclayton555 this is ready for you to take a look. I think the next task is to carry out the repository and config updates. For the updates:
- should I create a new issue or do we want to log the updates using this current issue?
- @aditigopalan do you have enough bandwidth to work through these steps or would it be helpful for me to do the updates?

**Comment by aditigopalan** [comment]
@Bankso sorry I missed this -- but happy to work through the steps! This way I can push all the data model changes together

**Comment by aditigopalan** [comment]
@Bankso please let me know if I missed anything! Changes in [this PR](https://github.com/Sage-Bionetworks/data_curator_config/pull/140)

**Comment by Bankso** [comment]
I wanted to propose some next steps for the current data model and DCA config work associated with this issue:

**Data model repo updates**

- I would like to propose we release a new version of our data model that includes the re-located dca-template-config.json, since it will need to be referenced in the config updates outlined below
- To prep for a release, I pulled together the dca-template-config.json and all the updates from the past few weeks, re-built the data model, and created excel templates for the manifests we use, per this updated pull request: https://github.com/mc2-center/data-models/pull/41 
- @aditigopalan if this seems reasonable and you have time, could you please review these changes? If everything checks out, would you be willing to release a new version? 

**Data curator config updates**

- After the data model release, we can update the DCA config excel templates (copy them over from data-models/templates) and URLs (data model JSON-LD and dca-template-config.json) using this [branch of the mc2-center/data_curator_config repo](https://github.com/mc2-center/data_curator_config/tree/refactor-data-models-42)
- Next, make a PR to merge updates into the mc2-center/data_curator_config prod branch
- From here, the mc2-center prod can be merged into Sage Bionetworks prod

@aclayton555 - please let me know if this seems like it should be logged separately, since it technically adds some work mid-sprint

**Comment by aditigopalan** [comment]
New version released [v4.1.0](https://github.com/mc2-center/data-models/releases/tag/v4.1.0) @Bankso feel free to edit the changelog if I missed anything!

**Comment by Bankso** [comment]
Looks good, thank you!

**Comment by aclayton555** [comment]
@aditigopalan and @Bankso if I understand correctly, I think we are still pending the steps for the Data curator config updates?

**Comment by aditigopalan** [comment]
I think we are all set per the last comment, @Bankso are we ready to sync into Sage Bionetworks prod?

> I wanted to propose some next steps for the current data model and DCA config work associated with this issue:
> 
> **Data model repo updates**
> 
> * I would like to propose we release a new version of our data model that includes the re-located dca-template-config.json, since it will need to be referenced in the config updates outlined below
> * To prep for a release, I pulled together the dca-template-config.json and all the updates from the past few weeks, re-built the data model, and created excel templates for the manifests we use, per this updated pull request: [24-1 model updates #41](https://github.com/mc2-center/data-models/pull/41)
> * @aditigopalan if this seems reasonable and you have time, could you please review these changes? If everything checks out, would you be willing to release a new version?
> 
> **Data curator config updates**
> 
> * After the data model release, we can update the DCA config excel templates (copy them over from data-models/templates) and URLs (data model JSON-LD and dca-template-config.json) using this [branch of the mc2-center/data_curator_config repo](https://github.com/mc2-center/data_curator_config/tree/refactor-data-models-42)
> * Next, make a PR to merge updates into the mc2-center/data_curator_config prod branch
> * From here, the mc2-center prod can be merged into Sage Bionetworks prod
> 
> @aclayton555 - please let me know if this seems like it should be logged separately, since it technically adds some work mid-sprint

**Comment by Bankso** [comment]
@aditigopalan yep, the last step is to make a PR for mc2-center/data_curator_config/prod to merge into Sage-Bionetworks/data_curator_config/prod

**Comment by Bankso** [comment]
@aditigopalan thanks for opening this [PR](https://github.com/Sage-Bionetworks/data_curator_config/pull/142)
Once merged, this issue can be closed

**Comment by Bankso** [comment]
PR has been merged: https://github.com/Sage-Bionetworks/data_curator_config/pull/142
